src/cloud-functions.ts

```diff
@@ -312,6 +320,41 @@ export interface MakeCloudFunctionArgs<EventData> {
   triggerResource: () => string;
 }
 
+/** @hidden */
+export function optionsToTrigger({
+  failurePolicy,
+  memory,
+  regions,
+  schedule,
+  timeoutSeconds,
+}: DeploymentOptions): TriggerAnnotated['__trigger'] {
+  const defaultFailurePolicy: FailurePolicy = {
+    retry: {},
+  };
+
+  const memoryLookup = {
+    '128MB': 128,
+    '256MB': 256,
+    '512MB': 512,
+    '1GB': 1024,
+    '2GB': 2048,
+  };
+
+  return {
+    ...(failurePolicy === undefined || failurePolicy === false
```

Comment:
I don't like the double use of the ternary operator here as it's quite confusing, it took me several pass throughs to understand what was happening with `failurePolicy`. Can we perhaps tackle this separately after we do the spread of all the other objects by tacking it on to the resulting object with some if statements?

Reply:
I hope this helps with readability: 7b56f21194a9415b3158713e97c74335ecae3bf3, let me know if that's sufficient.

Reply:
This looks great now, thanks